src/io/flutter/view/FlutterView.java

```diff
@@ -213,49 +201,53 @@ private void addInspector(FlutterApp app, @Nullable InspectorService inspectorSe
-      addInspectorPanel("Render Tree", runnerTabs, state, InspectorService.FlutterTreeType.renderObject, app, inspectorService, toolWindow,
-                        toolbarGroup, false,
-                        false);
+    if (debugConnectionAvailable) {
+      if (hasInspectorService) {
+        final boolean detailsSummaryViewSupported = inspectorService.isDetailsSummaryViewSupported();
+        runnerTabs.setSelectionChangeHandler(this::onTabSelectionChange);
+
+        addInspectorPanel(WIDGET_TAB_LABEL, runnerTabs, state, InspectorService.FlutterTreeType.widget, app, inspectorService, toolWindow,
+                          toolbarGroup, true, detailsSummaryViewSupported);
+        addInspectorPanel(RENDER_TAB_LABEL, runnerTabs, state, InspectorService.FlutterTreeType.renderObject, app, inspectorService,
+                          toolWindow,
+                          toolbarGroup, false, false);
+      }
+      else {
+        // If in profile mode, add disabled tabs for the inspector.
+        addDisabledTab(WIDGET_TAB_LABEL, runnerTabs, app, toolbarGroup);
+        addDisabledTab(RENDER_TAB_LABEL, runnerTabs, app, toolbarGroup);
+      }
+
+      addPerformanceTab(runnerTabs, app, !hasInspectorService);
     }
     else {
-      toolbarGroup.add(new OverflowAction(this, app));
-
-      final ActionToolbar toolbar = ActionManager.getInstance().createActionToolbar("InspectorToolbar", toolbarGroup, true);
-      final JComponent toolbarComponent = toolbar.getComponent();
-      toolbarComponent.setBorder(IdeBorderFactory.createBorder(SideBorder.BOTTOM));
-      tabContainer.add(toolbarComponent, BorderLayout.NORTH);
-
-      // Add a message about the inspector not being available in profile mode.
-      final JBLabel label = new JBLabel("Widget info not available in profile mode", SwingConstants.CENTER);
+      // Add a message about the inspector not being available in release mode.
+      final JBLabel label = new JBLabel("Inspector not available in release mode", SwingConstants.CENTER);
       label.setForeground(UIUtil.getLabelDisabledForeground());
       tabContainer.add(label, BorderLayout.CENTER);
     }
-
-    final boolean isVertical = !toolWindow.getAnchor().isHorizontal();
-    if (isVertical) {
-      final JPanel dashboardsPanel = new JPanel(new BorderLayout());
-      tabContainer.add(dashboardsPanel, BorderLayout.SOUTH);
-
-      if (FlutterSettings.getInstance().isShowHeapDisplay()) {
-        dashboardsPanel.add(FPSDisplay.createJPanelView(runnerTabs, app), BorderLayout.NORTH);
-        dashboardsPanel.add(HeapDisplay.createJPanelView(runnerTabs, app), BorderLayout.SOUTH);
-      }
-    }
   }
 
   private ActionCallback onTabSelectionChange(TabInfo info, boolean requestFocus, @NotNull ActiveRunnable doChangeSelection) {
-    final InspectorPanel panel = (InspectorPanel)info.getComponent();
-    panel.setVisibleToUser(true);
+    if (info.getComponent() instanceof InspectorTabPanel) {
```

Comment:
Aren't all the tabs now InspectorTabPanel objects?

Reply:
All but one; we could make that one an `InspectorTabPanel`, but w/o `TabInfo` being parametrizable, I like the instanceof checks; happy to change this however.